### PR TITLE
feat : Item class implementation #40

### PR DESCRIPTION
**Pull request by vivekmahindrakar**
Fixes : Item class implementation
issue : #40

### CODE REVIEW

**Comment by CoasterFreakDE** [comment]
Why do we need the extra layer of private and public variables?
If we use public variables in the constructor, Kotlin will do the rest for us. No need for additional variables.

Or was there an intention behind your decision for later use?

File: src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        get() = _translationKey
+        set(value) {
+            _translationKey = value
+        }
+
+    var maxItemStack: Int
+        get() = _maxItemStack
+        set(value) {
+            _maxItemStack = value
+        }
```
Comment (CoasterFreakDE) [inline src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt:20]: As the variables are all exposed, I would recommend to just expose them in the first place.
Also, i don't think we need to make those variables mutable.


```kotlin
data class Item(val name: String, val maxItemStack: Int = 99, val translationKey: String = "item.$name") {
```

This should do the job just fine

File: src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    // Utility methods can be added here
```
Comment (CoasterFreakDE) [inline src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt:22]: Maybe convert to javadoc?

```kotlin
/**
 * A function for using this item
 */
 ```

**Review by CoasterFreakDE — CHANGES_REQUESTED** [review]

File: src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt (hunk 1 above)
Comment (CoasterFreakDE) [reply src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt]: fixed

File: src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt (hunk 2 above)
Comment (CoasterFreakDE) [reply src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt]: fixed

**Review by CoasterFreakDE — APPROVED** [review]

File: src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt (hunk 1 above)
Comment (vivekmahindrakar) [reply src/main/kotlin/de/coasterfreak/fantasyfrontiers/data/model/items/Item.kt]: Okay i will do the required changes…thanks for the suggestion @CoasterFreakDE

**Pull request merged**